code/game/mecha/mecha.dm

```diff
@@ -502,8 +504,9 @@
 				check_for_internal_damage(list(MECHA_INT_TEMP_CONTROL,MECHA_INT_TANK_BREACH,MECHA_INT_CONTROL_LOST))
 			else
 				check_for_internal_damage(list(MECHA_INT_FIRE,MECHA_INT_TEMP_CONTROL,MECHA_INT_TANK_BREACH,MECHA_INT_CONTROL_LOST,MECHA_INT_SHORT_CIRCUIT))
-		if(. >= 5 || prob(33))
-			occupant_message("<span class='userdanger'>Taking damage!</span>")
+		if(. >= 5 || prob(33)) //If it takes 1 damage and leg_overload_mode is true, do not say TAKING DAMAGE! to the user several times a second.
+			if(!(. != 1 && leg_overload_mode))
```

Comment:
I think it should be `. == 1` because, taking the previous example:
Let `. = 10` and `leg_overload_mode = TRUE`
```dm
if(TRUE)
    if(!(TRUE && TRUE)) // <- if(!TRUE) does not pass, message is not shown.
```
Also you can still one-line this:
```dm
if((. >= 5 || prob(33)) && !(. == 1 && leg_overload_mode))
```